apps/api/src/report/report.service.ts

```diff
@@ -764,6 +785,15 @@ export class ReportService {
     const data = await entityManager.query(
       this.reportUtilService.extractApplicantsDataQuery(from, to, milestones),
     );
+
+    // set IEN type
+    const types = await this.getIenTypes(from, to);
```

Comment:
Why do we need a second call to fetch the types data? Shouldn't the types information be included in `data` variable?

Reply:
I couldn't find a proper way to do that. The query to get the `data` is already a little heavy, taking about 8 seconds, and I didn't want to make it more complex and heavy unless it is possible just by a select field, as I did for the milestone report.